wai-aria/heading/invalid-nesting.html

```diff
@@ -0,0 +1,216 @@
+    data-expectedrole="heading"
+  >
+    <h2
+      class="ex-role"
+      data-testname="invalid child h2 still conveys heading role"
+      data-expectedrole="heading"
+    >Malformed h2</h2>
+  </h1>
+
+  <div
+    role="heading"
+    aria-level="1"
+    class="ex-role"
+    data-testname="invalid parent aria-level=1 still conveys heading role"
+    data-expectedrole="heading"
+  >
+    <div
+      role="heading"
+      aria-level="2"
+      class="ex-role"
+      data-testname="invalid child aria-level=2 still conveys heading role"
+      data-expectedrole="heading"
+    >Malformed aria-level=2</div>
+  </div>
+
+  <h4
+    class="ex-role"
+    data-testname="invalid parent h4 still conveys heading role"
+    data-expectedrole="heading"
+  >
+    <h3
+      class="ex-role"
+      data-testname="invalid child h3 still conveys heading role"
+      data-expectedrole="heading"
+    >Malformed h3</h3>
+  </h4>
+
+  <div
+    role="heading"
+    aria-level="4"
+    class="ex-role"
+    data-testname="invalid parent aria-level=4 still conveys heading role"
+    data-expectedrole="heading"
+  >
+    <div
+      role="heading"
+      aria-level="3"
+      class="ex-role"
+      data-testname="invalid child aria-level=3 still conveys heading role"
+      data-expectedrole="heading"
+    >Malformed aria-level=3</div>
+  </div>
+
+  <h2>No heading ancestors</h2>
+  <hr>
+
+  <h1>One
+    <h2>Two
+      <h3
+        class="ex-no-heading-ancestor"
+        data-testname="nested h3 has no heading ancestor in DOM"
+      >Three</h3>
+    </h2>
+  </h1>
+
+  <div
+    role="heading"
+    aria-level="1"
+  >One
+    <div
+      role="heading"
+      aria-level="2"
+    >Two
+      <div
+        role="heading"
+        aria-level="3"
+        class="ex-no-heading-ancestor"
+        data-testname="nested aria-level=3 has no heading ancestor in DOM"
+      >Three</div>
+    </div>
+  </div>
+
+  <script>
+    AriaUtils.verifyLabelsBySelector(".ex-label");
+    AriaUtils.verifyRolesBySelector(".ex-role");
+    verifyNoAncestorRole(".ex-no-heading-ancestor", "heading");
+
+    function verifyNoAncestorRole(selector, forbiddenRole) {
+      const els = document.querySelectorAll(selector);
+      if (!els.length) {
+        throw `Selector passed in verifyNoAncestorRole("${selector}") should match at least one element.`;
+      }
+      for (const el of els) {
+        let testName = el.getAttribute("data-testname");
+        let ancestors = [];
+        let ancestor = el.parentNode;
+        promise_test(async t => {
+          // Get computed role of each ancestor up to but excluding the root.
+          while (ancestor && ancestor != document.documentElement) {
+            const computedRole = await test_driver.get_computed_role(ancestor);
+            ancestors.push(computedRole);
+            ancestor = ancestor.parentNode;
+          }
+          assert_true(!ancestors.includes(forbiddenRole), el.outerHTML);
```

Comment:
Nit: `assert_true !not` is awkward to read... Check out the WPT source where that method defined an see if there are others related to containment, or asserting false instead of asserting _not_ true?